contracts/l2/staking/L2Staking.sol

```diff
@@ -122,7 +125,14 @@ contract L2Staking is Staking, IL2StakingBase {
         Delegation storage delegation = pool.delegators[_delegationData.delegator];
 
         // Calculate shares to issue (without applying any delegation tax)
-        uint256 shares = (pool.tokens == 0) ? _amount : _amount.mul(pool.shares).div(pool.tokens);
+        uint256 shares;
+        if (pool.tokens == 0) {
+            if (_amount >= MINIMUM_DELEGATION) {
```

Comment:
I would add a minimal comment explaining or referencing the issue this fixes. It's not clear why we are doing this otherwise.